### PR TITLE
Song liking and history visualization

### PR DESCRIPTION
**Pull request by Alberto-Cortes**
- Songs are recommended based on the captured image and shown in a stack of cards that can be swiped to like or reject each song.
- User is now able to liked recommended songs by swiping to the right the card presented after taking a photo.
- Activities were sent to a directory that only holds the Java classes for the Activities of the project, in a similar manner to the fragments.
- Songs that had been liked are now presented on a RecyclerView to the user where each row is a Material Card, tapping anywhere on the card will lead to a song detail view.
- Login and signup were updated in a way that new users and users that have not completed their ColorForm will always be sent to the ColorFormActivity before entering the rest of the app.

### CODE REVIEW

File: app/src/main/java/com/example/photostomusic/Activities/ColorFormActivity.java [hunk 1]
```diff
@@ -38,6 +39,11 @@
     AppCompatSpinner spinner6;
     AppCompatSpinner spinner7;
     AppCompatSpinner spinner8;
+    String spotifyToken = "";
+
+    // HEX number used to remove Alpha values from colors as it will not be used. This as Android
+    // reads color by default as ARGB, only RGB is needed.
+    public static final  int HEX_BASE = 0x00ffffff;
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/Activities/ColorFormActivity.java:46]: nice!

File: app/src/main/java/com/example/photostomusic/Activities/ColorFormActivity.java [hunk 2]
```diff
@@ -84,14 +92,14 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        String color7 = Integer.toHexString(getResources().getColor(R.color.option7) & 0x00ffffff);
-        String color8 = Integer.toHexString(getResources().getColor(R.color.option8) & 0x00ffffff);
+        String color1 = Integer.toHexString(getResources().getColor(R.color.option1) & HEX_BASE);
+        String color2 = Integer.toHexString(getResources().getColor(R.color.option2) & HEX_BASE);
+        String color3 = Integer.toHexString(getResources().getColor(R.color.option3) & HEX_BASE);
+        String color4 = Integer.toHexString(getResources().getColor(R.color.option4) & HEX_BASE);
+        String color5 = Integer.toHexString(getResources().getColor(R.color.option5) & HEX_BASE);
+        String color6 = Integer.toHexString(getResources().getColor(R.color.option6) & HEX_BASE);
+        String color7 = Integer.toHexString(getResources().getColor(R.color.option7) & HEX_BASE);
+        String color8 = Integer.toHexString(getResources().getColor(R.color.option8) & HEX_BASE);
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/Activities/ColorFormActivity.java:102]: could this be pulled into a separate function? it seemed like a lot of repeated code

File: app/src/main/java/com/example/photostomusic/Activities/LoginActivity.java [hunk 3]
```diff
@@ -1,30 +1,38 @@
...
+import java.util.HashMap;
+import java.util.Objects;
+
 public class LoginActivity extends AppCompatActivity {
 
     // Class name used as TAG for debugging
     public final String TAG = this.getClass().getSimpleName();
 
     // PSF variables required to interact with the Spotify Auth API
+    // TODO: Hide keys and IDs on properties file
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/Activities/LoginActivity.java:33]: i'd do this ASAP

File: app/src/main/java/com/example/photostomusic/fragments/CameraFragment.java [hunk 4]
```diff
@@ -161,14 +163,14 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
...
-        String color7 = Integer.toHexString(getResources().getColor(R.color.option7) & 0x00ffffff);
-        String color8 = Integer.toHexString(getResources().getColor(R.color.option8) & 0x00ffffff);
+        String color1 = Integer.toHexString(getResources().getColor(R.color.option1) & HEX_BASE);
+        String color2 = Integer.toHexString(getResources().getColor(R.color.option2) & HEX_BASE);
+        String color3 = Integer.toHexString(getResources().getColor(R.color.option3) & HEX_BASE);
+        String color4 = Integer.toHexString(getResources().getColor(R.color.option4) & HEX_BASE);
+        String color5 = Integer.toHexString(getResources().getColor(R.color.option5) & HEX_BASE);
+        String color6 = Integer.toHexString(getResources().getColor(R.color.option6) & HEX_BASE);
+        String color7 = Integer.toHexString(getResources().getColor(R.color.option7) & HEX_BASE);
+        String color8 = Integer.toHexString(getResources().getColor(R.color.option8) & HEX_BASE);
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/fragments/CameraFragment.java:173]: same here, lots of repeated code that could be pulled out. in fact, the HEX_BASE and this Int.toHex stuff could all be pulled out into more reusable code

File: app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java [hunk 5]
```diff
@@ -71,57 +84,192 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+                            return true;
+                        }
+
+                        @Override
+                        // TODO: Check for direction with switch case, push song to Parse if 1 or 3 were selected, next song if 0 or 2 were selected.
+                        public void discarded(int mIndex, int direction) {
+                            // there are four directions
+                            //  0  |  1
+                            // ----------
+                            //  2  |  3
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java:246]: great comment!

**Review by mbarbagallo — APPROVED** [review]
Looks good. Let's make sure we get rid of the duplicate code I mentioned.

File: app/src/main/java/com/example/photostomusic/Activities/LoginActivity.java (hunk 3 above)
Comment (Alberto-Cortes) [reply app/src/main/java/com/example/photostomusic/Activities/LoginActivity.java:33]: Done :)
Will be pushed on next PR

File: app/src/main/java/com/example/photostomusic/Activities/ColorFormActivity.java (hunk 2 above)
Comment (Alberto-Cortes) [reply app/src/main/java/com/example/photostomusic/Activities/ColorFormActivity.java:102]: Fixed, this is now done in a for each loop.

File: app/src/main/java/com/example/photostomusic/fragments/CameraFragment.java (hunk 4 above)
Comment (Alberto-Cortes) [reply app/src/main/java/com/example/photostomusic/fragments/CameraFragment.java:173]: Done :)

**Pull request merged**